graphtools.py

```diff
@@ -55,13 +55,17 @@ def DFS(graph, processingFx = print):
 def bfs(graph, marked, vertex, processingFx):
     processingFx(vertex)
     marked[vertex] = True
+    queue = [vertex]
```

Comment:
There is an actual queue object in the form of collections.deque. Try using that. Doing it with a list leads to O(n) deletion time